common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/WorkerSupervisor.cs

```diff
@@ -39,6 +41,9 @@ public class WorkerSupervisor : IWorkerSupervisor, IDisposable {
             _logger = logger;
             _ensureWorkerRunningTimer = new Timer(TimeSpan.FromSeconds(timerDelayInSeconds).TotalMilliseconds);
             _ensureWorkerRunningTimer.Elapsed += EnsureWorkerRunningTimer_ElapsedAsync;
+            _agentConfigProvider.OnConfigUpdated += (s, e) => {
```

Comment:
We need to clean up the callbacks that we added on `Dispose()`. That applies for `_agentConfigProvider.OnConfigUpdated` and maybe also for `_ensureWorkerRunningTimer.Elapsed`.

Reply:
done